pyanaconda/startup_utils.py

```diff
@@ -646,3 +646,36 @@ def check_if_geolocation_should_be_used(opts):
 
     log.info("Geolocation is enabled.")
     return True
+
+
+def start_geolocation_conditionally(opts):
+    """Start geolocation conditionally, according to the command line or boot options.
+
+    :param opts: the command line/boot options
+    :return: a DBus path of the geolocation task
+    """
+    use_geoloc = check_if_geolocation_should_be_used(opts)
+    if not is_module_available(TIMEZONE) or not use_geoloc:
+        return None
+
+    timezone_proxy = TIMEZONE.get_proxy()
+    geoloc_task_path = timezone_proxy.StartGeolocationWithTask()
+    geoloc_task_proxy = TIMEZONE.get_proxy(geoloc_task_path)
+    geoloc_task_proxy.Start()
+    return geoloc_task_proxy
+
+
+def wait_for_geolocation(geoloc_task_proxy):
+    """Wait for geolocation, if started
+
+    :param geoloc_task_proxy: path to the GeolocationTask proxy
```

Comment:
It accepts a DBus proxy. DBus path is a string. DBus proxy is a wrapper of a remote object that is specified by the DBus path.